### ISSUE TITLE
Add SigLip loss

### ISSUE THREAD

**Issue by robertavram-md** [HeartWise-AI/DeepCORO_CLIP#39]
```import torch
import torch.nn.functional as F

def siglip_loss(image_embeddings, text_embeddings, temperature=1.0):
    """
    Compute the SigLIP loss for a batch of image and text embeddings.

    Args:
        image_embeddings (torch.Tensor): Tensor of shape (batch_size, embedding_dim) containing image embeddings.
        text_embeddings (torch.Tensor): Tensor of shape (batch_size, embedding_dim) containing text embeddings.
        temperature (float): Scaling factor for the logits. Default is 1.0.

    Returns:
        torch.Tensor: Scalar tensor representing the SigLIP loss.
    """
    # Normalize embeddings
    image_embeddings = F.normalize(image_embeddings, p=2, dim=-1)
    text_embeddings = F.normalize(text_embeddings, p=2, dim=-1)

    # Compute pairwise similarity scores
    logits = torch.matmul(image_embeddings, text_embeddings.t()) / temperature

    # Create labels: 1 for positive pairs, 0 for negative pairs
    batch_size = image_embeddings.size(0)
    labels = torch.eye(batch_size, device=logits.device)

    # Compute binary cross-entropy loss with logits
    loss = F.binary_cross_entropy_with_logits(logits, labels)

    return loss```
